crates/node/src/shell/process_proposal.rs

```diff
@@ -925,83 +926,85 @@ mod test_process_proposal {
     /// Test that a block including a wrapper tx with invalid signature is
     /// rejected
     #[test]
-    fn test_wrapper_bad_signature_rejected() {
+    fn test_wrapper_bad_signature() {
         let (shell, _recv, _, _) = test_utils::setup_at_height(3u64);
-        let keypair = gen_keypair();
-        let mut outer_tx =
-            Tx::from_type(TxType::Wrapper(Box::new(WrapperTx::new(
-                Fee {
-                    amount_per_gas_unit: DenominatedAmount::native(
-                        Amount::from_uint(100, 0).expect("Test failed"),
-                    ),
-                    token: shell.state.in_mem().native_token.clone(),
-                },
-                keypair.ref_to(),
-                GAS_LIMIT.into(),
-            ))));
-        outer_tx.header.chain_id = shell.chain_id.clone();
-        outer_tx.set_code(Code::new("wasm_code".as_bytes().to_owned(), None));
-        outer_tx.set_data(Data::new("transaction data".as_bytes().to_owned()));
-        outer_tx.sign_wrapper(keypair);
-        let mut new_tx = outer_tx.clone();
-        if let TxType::Wrapper(wrapper) = &mut new_tx.header.tx_type {
-            // we mount a malleability attack to try and remove the fee
-            wrapper.fee.amount_per_gas_unit =
-                DenominatedAmount::native(Default::default());
-        } else {
-            panic!("Test failed")
-        };
-        let request = ProcessProposal {
-            txs: vec![new_tx.to_bytes()],
-        };
 
-        match shell.process_proposal(request) {
-            Ok(_) => panic!("Test failed"),
-            Err(TestError::RejectProposal(response)) => {
-                let response = if let [response] = response.as_slice() {
-                    response.clone()
-                } else {
-                    panic!("Test failed")
-                };
-                let expected_error = "WrapperTx signature verification \
-                                      failed: The wrapper signature is \
-                                      invalid.";
-                assert_eq!(
-                    response.result.code,
-                    u32::from(ResultCode::InvalidSig)
-                );
-                assert!(
-                    response.result.info.contains(expected_error),
-                    "Result info {} doesn't contain the expected error {}",
-                    response.result.info,
-                    expected_error
-                );
+        let mut runner = TestRunner::new(Config::default());
```

Comment:
I'm doing this to avoid rerunning the setup for every iteration but I'm not completely familiar with the implications of instantiating a test runner over just calling `protest!`, in case I can switch to the latter

Reply:
I think that's fine yeah - it's pretty much what the macro expands to